core/src/apps/binance/sign_tx.py

```diff
@@ -0,0 +1,65 @@
+from trezor.crypto.curve import secp256k1
+from trezor.crypto.hashlib import sha256
+from trezor.messages import MessageType
+from trezor.messages.BinanceSignedTx import BinanceSignedTx
+from trezor.messages.BinanceTxRequest import BinanceTxRequest
+
+from apps.binance import CURVE, helpers, layout
+from apps.common import paths
+
+
+async def sign_tx(ctx, envelope, keychain):
+    # create transaction message -> sign it -> create signature/pubkey message -> serialize all
+    await paths.validate_path(
+        ctx, helpers.validate_full_path, keychain, envelope.address_n, CURVE
+    )
+
+    node = keychain.derive(envelope.address_n)
+
+    tx_req = BinanceTxRequest()
+
+    msg = await ctx.call(
```

Comment:
I can add a loop however based on the documentation and how their SDK is designed it seems msg_count will always equal 1
https://docs.binance.org/encoding.html#standard-transaction
https://github.com/binance-chain/javascript-sdk/blob/master/src/client/index.js#L237

Reply:
to support msg_count>1 we would also have to modify the BinanceSignedTx message to return multiple signatures.

Reply:
thinking about it more - BinanceSignTx would have to have a different sequence value for each message. At that point I think looping over the messages and sending them for signing should be responsibility of client and we should raise an error if msg_count > 1

Reply:
[facepalms]
ok
well one option is to enforce that `msg_count` is set to 1
another is to revert to an earlier design variant, where the message type was inlined in the `ProtobufSignTx` message, and get rid of `msg_count` altogether.

please do whichever you think fits the Binance SDK better

Reply:
... actually, let's keep the current design and enforce msg_count is 1. otherwise in the future Binance will allow more messages and someone will try to covert the optional field to a repeated field, which is what we wanted to avoid in the first place

Reply:
Ok. I raise error in case of msg_count > 1 in firmware and trezorctl sets it to 1 and ignores other messages the way it is now.